MetaMask/metamask-mobile
# feat: add SES experiment toggle (iOS)

## **Description**



A feature toggle for SES, within our settings menu under experiments, enabled by default

TODO
- [x] add `react-native-mmkv`
- [x] MVP
- [x] ensure only visible on iOS not Android
  - show only on iOS
  - or show if storage key exists
- [x] check next to Blockaid experiment enabled
- [x] fix test, update snapshot
- [x] add descriptive content
- [x] translate text
- [x] update styling
- [x] <s>replace `react-native-mmkv` with `react-native-default-preference`</s>
  - attempted in https://github.com/MetaMask/metamask-mobile/pull/8373/commits/ea5674c5600d5d11d78a07ff4ab774cf3848989f but only [partially working](https://github.com/MetaMask/metamask-mobile/pull/8373#issuecomment-1908544768)
- [x] fix Android debug bundling on Hermes
- [x] <s>consider content revision to include _Hardened JS_  and _SES (Secure EcmaScript)_ terms</s>
- [x] update to newly discussed text with link
- [x] fix styling, move switch above next to title
- [x] display both SES and Blockaid under same _Security_ heading
- [x] preserve Android UI
- [x] update SES to be enabled by default

## **Related issues**

Fixes: partially mitigates rollout risk

Follow-up to
- https://github.com/MetaMask/metamask-mobile/pull/8033

## **Manual testing steps**

iOS
- boot app
- login / create wallet
- go to bottom tab Settings menu
- open Experimental menu
- _SES enabled by default, like currently in (main) branch_
- disable SES ft toggle
- reboot app
- _SES now disabled_
- verify in Experimental menu
- repeat steps above to re-enable

Android
- boot app
- login / create wallet
- go to bottom tab Settings menu
- open Experimental menu
- UI remains the same
  - Blockaid shown if feature is enabled

## **Screenshots/Recordings**

Previous MVP video (old UI) [demo](https://consensys.slack.com/archives/C02U025CVU4/p1706035121992709?thread_ts=1705527596.738189&cid=C02U025CVU4) showing the behaviour, a simple settings menu ft toggle that persists, enabling/disabling lockdown upon app reboot, on iOS only

Android remains unchanged ([Blockaid disabled](https://github.com/MetaMask/metamask-mobile/assets/1881059/d74a30be-e65e-4832-a21e-8aa1812df608), [Blockaid enabled](https://github.com/MetaMask/metamask-mobile/assets/1881059/77b9a24b-486b-4e6b-aa10-51eede8c2951))

### **Before**

Blockaid ft disabled

<img width="403" alt="Screenshot 2024-01-26 at 7 20 46 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/ec120e0a-9ba5-454c-b4c8-21307ced7ea5">

Blockaid ft enabled

<img width="402" alt="Screenshot 2024-01-26 at 7 25 05 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/4be5fda0-edfe-427f-8206-df6df100a6fc">

### **After**

Blockaid ft disabled

<img width="403" alt="Screenshot 2024-01-30 at 1 07 07 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/bca71f83-ab01-4d70-bbc1-fac00dd76e23">

Blockaid ft enabled

<img width="401" alt="Screenshot 2024-01-30 at 1 07 48 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/7d12aa9a-51e9-44d8-a6d0-6d9ca8aa9375">

Navigating to [Learn more](https://github.com/endojs/endo/blob/master/packages/ses/README.md)

<img width="401" alt="Screenshot 2024-01-26 at 6 52 17 pm" src="https://github.com/MetaMask/metamask-mobile/assets/1881059/a32d47b3-e95e-4a15-b745-3cbc7c444250">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
